### ISSUE TITLE
interpreting PhiNodes may be broken

### ISSUE THREAD

**Issue by vtjnash** [JuliaLang/julia#29342]
It seems like it can encounter the same issue as 
https://github.com/JuliaLang/julia/commit/a5c119f172fe5ac6fe7a14c73424824fe71021af#diff-6d4d21428a67320600faf5a1a9f3a16a:
> [NewOptimizer] Refactor phinode codegen
Turns out the old approach we used didn't work, because it was possible
for the processing of one phi node to clobber the operands of another (e.g.
in a loop where one phi node depends on another). Fix this by using two
buffers. This isn't great, but hopefully LLVM should do ok at it. If
not, we can fix LLVM or try to be smarter.

with the current representation chosen, it's invalid to treat phi nodes sequentially as statements—they represent a `let` not a `let*`—so they must be always handled as a single block entity to avoid potentially clobbering the operands of another node.

**Comment by Keno** [comment]
Yes, I think you're right about that. We can probably fix this by finding all the phi nodes (as soon as we see an operation that's not a phi or `nothing`, we're guaranteed to have seen all the phis in this basic block) and allocating a temporary stack buffer before copying the operands into the result slots.
